### ISSUE TITLE
arm pagesize64k

### ISSUE THREAD

**Issue by gxiaob** [AndreRH/hangover#130]
does hangover work on arm server，pagesize 64k？

**Comment by stefand** [comment]
Most likely not. We never tested page sizes other than 4k

**Comment by darkbasic** [comment]
I've never used hangover (waiting for ppc64le support) but I'm pretty sure it doesn't because it relies on qemu which does a terrible job at emulating different page sizes. The Linux kernel should really support multiple page sizes like MacOS does.

**Comment by darkbasic** [comment]
Also, which arm chips do have a 64K page size? I know M1 chips use 16K.

**Comment by gxiaob** [comment]
Thank you！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "AndreRH/hangover"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年3月10日(星期五) 下午4:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [AndreRH/hangover] arm pagesize64k (Issue #130)





 
Also, which arm chips do have a 64K page size? I know M1 chips use 16K.

**Comment by darkbasic** [comment]
By the way most chips nowadays support multiple page sizes, so using 4K should be as wasy as recompiling your kernel with a different config. Performance penalties from 64K->4K are minimal.

**Comment by rafael2k** [comment]
Indeed. When I got a Raspberry Pi 5, nothing was working, and it was because of the 16k page size. Switching the kernel to 4k page size solved the problems.

**Comment by stefand** [comment]
There is some discussion on this topic in issue 97. I posted a comment speculating on things that can be done in Wine at https://github.com/AndreRH/hangover/issues/97#issuecomment-1580168792 . Needless to say, none of this is easy.

**Comment by AndreRH** [comment]
There were some adjustments regarding other page sizes in Wine lately, please retest with Hangover 10.6 to see if it works now.

**Comment by AndreRH** [comment]
AFAIK this is fixed

**Comment by rafael2k** [comment]
Wow, I did not know about it! Sorry the delay to check about it. Do you know of any performance penalty of running on 64k pagesize (stock windows PE 32 bits)?

**Comment by AndreRH** [comment]
Performance should be the same, but not every program will run
